app/repos/cdo.js

```diff
@@ -441,6 +444,59 @@ const getCdoTaskList = async (indexNumber) => {
  * }} CdoRepository
  */
 
+const updateMappings = {
+  applicationPackSent: 'dog.registration.application_pack_sent',
+  applicationFeePaid: 'dog.registration.application_fee_paid',
+  form2Sent: 'dog.registration.form_two_sent',
+  neuteringConfirmation: 'dog.registration.neutering_confirmation',
+  microchipVerification: 'dog.registration.microchip_verification',
+  certificateIssued: 'dog.registration.certificate_issued',
+  insuranceCompany: new Error('Not implemented'),
+  insuranceRenewalDate: new Error('Not implemented'),
+  microchipNumber: new Error('Not implemented')
+}
+
+/**
+ * @param {import('../data/domain/cdoTaskList').CdoTaskList} cdoTaskList
+ * @param transaction
+ * @return {Promise<void>}
+ */
+const saveCdoTaskList = async (cdoTaskList, transaction) => {
+  if (!transaction) {
+    return await sequelize.transaction(async (t) => saveCdoTaskList(cdoTaskList, t))
+  }
+  const updates = cdoTaskList.getUpdates()
+  const cdoDao = await getCdo(cdoTaskList.cdoSummary.indexNumber)
+  let preAudit
+  const preChangedExemptionAuditRecord = preChangedExemptionAudit(cdoDao)
+
+  for (const update of updates.exemption) {
```

Comment:
I may have missed something here but won;t this loop create multiple DB commits and multiple audit events for something like the both verification dates being changed, rather than a single DB save and a single audit event?